docs/examples/201/policy-definition-with-assignment/main.bicep

```diff
@@ -0,0 +1,77 @@
+targetScope = 'subscription'
+
+param listofAllowedLocations array {
+  default: [
+    'norwayeast'
+    'westeurope'
+  ]
+}
+param policyEffect string {
+  allowed:[
+    'Audit'
+    'Deny'
+  ]
+}
+
+param assignmentScope string
+
+resource locationPolicyDefinition 'Microsoft.Authorization/policyDefinitions@2020-09-01' = {
+  name: 'custom-allowed-location'
+  properties:{
+    displayName: 'Custom - allowed location for resources'
+    policyType: 'Custom'
+    description: 'Use policy to restrict where resources can be deployed'
+    parameters: {
+      allowedLocations: {
+        type: 'Array'
+      }
+      effect: {
+        type: 'String'
+      }
+    }
+    metadata: {
+      category: 'Locations'
+    }
+    policyRule: {
+      if: {
+        allOf: [
+          {
+            field: 'location'
+            notIn: '[parameters(\'allowedLocations\')]'
+          }
+          {
+            field: 'location'
+            notEquals: 'global'
+          }
+          {
+            field: 'type'
+            notEquals: 'Microsoft.AzureActiveDirectory/b2cDirectories'
+          }
+      ]
+    }
+      then: {
+        effect: '[parameters(\'effect\')]'
+      }
+    }
+  }
+}
+
+resource locationPolicy 'Microsoft.Authorization/policyAssignments@2020-09-01' = {
+ name: 'Resource-location-restriction'
+ dependsOn: [

```

Comment:
you can get rid of this `dependsOn`. You already have an implict dependency because you reference `locationPolicyDefinition.id` on line 65, so the will generate the template with the dependsOn correctly on your behalf.

More info here:
https://github.com/Azure/bicep/blob/8c810eab31ac5dbd011d55cfbf644068d3a2ff6b/docs/spec/resources.md#resource-dependencies